internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -827,13 +837,38 @@ func NewKeystoreSecretForCR(cr *operatorv1beta1.Cryostat) *corev1.Secret {
 	}
 }
 
-func NewServiceAccountForCR(cr *operatorv1beta1.Cryostat) *corev1.ServiceAccount {
+func NewServiceAccountForCR(cr *operatorv1beta1.Cryostat, isOpenShift bool) (*corev1.ServiceAccount, error) {
+	annotations := make(map[string]string)
+
+	if isOpenShift {
+		OAuthRedirectReference := &oauthv1.OAuthRedirectReference{
+			Reference: oauthv1.RedirectReference{
+				Kind: "Route",
+				Name: cr.Name,
+			},
+		}
+
+		ref, err := json.Marshal(OAuthRedirectReference)
+		if err != nil {
+			return nil, err
+		}
+
+		annotations = map[string]string{
+			"serviceaccounts.openshift.io/oauth-redirecturi.route":       "https://",
```

Comment:
I think perhaps we should remove this annotation. Right now the operator always creates Routes with some form of TLS enabled, but I think in the near future we may want to allow the user to customize the Routes we create like we do with Ingresses and now Services. In the interest of future-proofing this, I suggest we don't assume HTTPS here.

I tracked down the code that parses the route provided in the `oauth-redirectreference.route` annotation:
https://github.com/openshift/library-go/blob/7a65fdb398e28782ee1650959a5e0419121e97ae/pkg/oauth/oauthserviceaccountclient/oauthclientregistry.go#L418-L441

It will use HTTPS for the scheme, unless the Route has no TLS configuration, which sounds like what we'd want. We should be fine to not override this behaviour by forcing HTTPS.